internal/services/datafactory/data_factory_dataset_cosmosdb_mongoapi_resource.go

```diff
@@ -0,0 +1,339 @@
+package datafactory
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/datafactory/2018-06-01/factories"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+	"github.com/jackofallops/kermit/sdk/datafactory/2018-06-01/datafactory"
+)
+
+var _ sdk.Resource = DataFactoryDatasetCosmosDbMongoDbResource{}
+
+type DataFactoryDatasetCosmosDbMongoDbResource struct{}
+
+type DataFactoryDatasetCosmosDbMongoDbResourceModel struct {
+	AdditionalProperties map[string]interface{} `tfschema:"additional_properties"`
+	Annotations          []string               `tfschema:"annotations"`
+	CollectionName       string                 `tfschema:"collection_name"`
+	DataFactoryId        string                 `tfschema:"data_factory_id"`
+	Description          string                 `tfschema:"description"`
+	Folder               string                 `tfschema:"folder"`
+	LinkedServiceName    string                 `tfschema:"linked_service_name"`
+	Name                 string                 `tfschema:"name"`
+	Parameters           map[string]interface{} `tfschema:"parameters"`
```

Comment:
Probably we can change it to `map[string]string`?

Reply:
Fixed in 1179a4200a